static/src/javascripts-legacy/projects/common/modules/tailor/tailor-survey-overlay.js

```diff
@@ -0,0 +1,115 @@
+    'raw-loader!common/views/tailor-survey-overlay.html',
+    'common/views/svgs',
+    'ophan/ng'
+], function(
+    fastdomPromise,
+    config,
+    detect,
+    cookies,
+    template,
+    storage,
+    tailor,
+    tailorSurveyOverlayTemplate,
+    svgs,
+    ophan
+) {
+
+    var EDITION_CONFIG = {
+        default: {
+            copy: 'Take part in a survey for the Guardian and win Amazon vouchers',
+            url: 'http://sunesis.equinoa.com/1703/BLU/cgi-bin/ciwweb.pl?hid_studyname=BLU&v=c'
+        },
+        AU: {
+            copy: 'We would like you to take part in a short Guardian survey',
+            url: 'http://sunesis.equinoa.com/1703/BLUau/cgi-bin/ciwweb.pl?hid_studyname=BLUau&v=c'
+        }
+    };
+
+    function shouldCallTailor(forceShow) {
+        return  !config.page.shouldHideAdverts &&
+                !config.page.isSensitive &&
+                !config.page.isFront &&
+                (!cookies.get('GU_TAILOR_SURVEY_OVERLAY') || forceShow) &&
+                detect.getBreakpoint() !== 'mobile';
+    }
+
+    function handleResponse(showSurvey) {
+        if (showSurvey) {
+            return fastdomPromise.write(function () {
+                var surveyOverlay = document.createElement('div');
+
+                surveyOverlay.classList.add('tailor-survey-overlay');
+                surveyOverlay.dataset.component = 'tailor-survey-overlay';
+
+                surveyOverlay.innerHTML = template(tailorSurveyOverlayTemplate, {
+                    headerCopy: getEditionConfigProp('copy'),
+                    chevronRight: svgs('chevronRight'),
+                    cross: svgs('crossIcon'),
+                    marque36icon: svgs('marque36icon')
+                });
+
+                document.body.appendChild(surveyOverlay);
+
+                setupButtonClickEvents(surveyOverlay);
+            });
+        }
+    }
+
+    function setupButtonClickEvents(surveyOverlay) {
+        var proceedButton = surveyOverlay.querySelector('.tailor-survey-proceed-button');
+        var closeButton = surveyOverlay.querySelector('.tailor-survey-close-button');
+        var surveyURL = getEditionConfigProp('url');
+
+        if (config.ophan && config.ophan.pageViewId) {
+            surveyURL += '&id=' + config.ophan.pageViewId;
+        }
+
+        proceedButton.addEventListener('click', function () {
+            window.location = surveyURL; 
+        });
+
+        closeButton.addEventListener('click', function () {
+           surveyOverlay.parentNode.removeChild(surveyOverlay); 
+        });
+    }
+
+    function getEditionConfigProp(key) {
+        if (EDITION_CONFIG[config.page.edition]) {
+            return EDITION_CONFIG[config.page.edition][key];
+        } else {
+            return EDITION_CONFIG['default'][key];
+        }
+    }
+
+    function onSurveyAdded() {
+        ophan.record({
+            component: 'tailor-survey-overlay',
+            value: 'impression'
+        });
```

Comment:
Yeah this is something I've been trying to find examples of how this has been done elsewhere. These mediator calls will eventually just record 'register:begin' and 'register:end' with ophan in the same way I've directly recorded 'impression'. It doesn't feel quite right calling 2 events to register a single event (element added), but as this has been used elsewhere in this way I'm happy to do this here for now. I've set the data-component attribute in the JS.